Framework/Core/include/Framework/ASoA.h

```diff
@@ -125,9 +125,16 @@ class ColumnIterator : ChunkingPolicy
       mFirstIndex{0},
       mCurrentChunk{0}
   {
-    auto array = std::static_pointer_cast<arrow_array_for_t<T>>(mColumn->chunk(mCurrentChunk));
-    mCurrent = reinterpret_cast<T const*>(array->values()->data()) + array->offset();
-    mLast = mCurrent + array->length();
+    auto chunkToUse = getCurrentChunk();
+    if constexpr (std::is_same_v<arrow_array_for_t<T>, arrow::FixedSizeListArray>) {
+      auto array = std::static_pointer_cast<arrow_array_for_t<element_for_t<T>>>(chunkToUse);
```

Comment:
Why not having the cast as part of `getCurrentChunk`? You could then completely forget about the if constexpr.